src/client/shadows/dynamicshadowsrender.cpp

```diff
@@ -322,9 +322,10 @@ void ShadowRenderer::update(video::ITexture *outputTarget)
 	if (!m_shadow_node_array.empty() && !m_light_list.empty()) {
 
 		for (DirectionalLight &light : m_light_list) {
-			// Static shader values.
-			m_shadow_depth_cb->MapRes = (f32)m_shadow_map_texture_size;
-			m_shadow_depth_cb->MaxFar = (f32)m_shadow_map_max_distance * BS;
+			// Static shader values for entities are set in updateSMTextures
+			// SM texture for entities is not updated incrementally and 
+			// must by updated using current player position.
+			m_shadow_depth_entity_cb->CameraPos = light.getPlayerPos();
```

Comment:
What's the difference between: getPlayerPos(), getFuturePlayerPos() (and also worldPosition)?

Reply:
This is related to changes in #11422, where I added two light frustums: active for rendering the shadows in the world and "future" for rendering the shadowmap incrementally over several frames. Since player position is now a key parameter for shadow mapping, it also exists in 'active' and 'future' versions.